.github/s2n_bsd.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+set -eu
+export CTEST_OUTPUT_ON_FAILURE=1
+
+if [ "$(uname)" = 'FreeBSD' ]; then
+    export CTEST_PARALLEL_LEVEL=$(sysctl hw.ncpu | awk '{print $2}')
+else
+    export CTEST_PARALLEL_LEVEL=$(sysctl -n hw.ncpuonline)
+fi
+
+errors=0
+onerror() {
+  errors=$(($errors+1))
+}
+
+mkdir -p output
+
+cmake . -Brelease -GNinja -DCMAKE_BUILD_TYPE=Release || onerror
```

Comment:
`|| onerror` makes the script continue on when there is an error, even though we have `set -e` at the top. I'd prefer the script stop immediately, as after an error the subsequent lines errors won't be meaningful. I think it makes sense to limit `|| onerror` to the lines that run the tests.

Reply:
i can do that. I did it this way to maximize the chance that you get _some_ kind of artifact out of it but it's not clear to me that any of the test artifacts will actually aid in debugging a failure or not